mojave/items/guns/energy.dm

```diff
@@ -50,46 +134,88 @@
 	icon = 'mojave/icons/objects/guns/guns_world.dmi'
 	inhand_icon_state = "plasmacarabine"
 	icon_state = "plasmacarabine"
+	base_icon_state = "plasmacarabine"
 	lefthand_file = 'mojave/icons/mob/inhands/weapons/guns_inhand_left.dmi'
 	righthand_file = 'mojave/icons/mob/inhands/weapons/guns_inhand_right.dmi'
 	desc = "A top of line miniaturized plasma caster built by REPCONN in the wake of the Z43-521P failure. It is supperior to all previous rifles to enter service in the USCC."
 	ammo_type = list(/obj/item/ammo_casing/energy/ms13/plasma)
-	cell_type = /obj/item/stock_parts/cell/ammo/mfc
+	cell_type = /obj/item/stock_parts/cell/ammo/pc
 	w_class = WEIGHT_CLASS_HUGE
 
 /obj/item/gun/energy/ms13/laser/plasma/scatter
 	name = "A3e-20b Multiplas Rifle"
 	icon = 'mojave/icons/objects/guns/guns_world.dmi'
 	inhand_icon_state = "multiplas"
 	icon_state = "multiplas"
+	base_icon_state = "multiplas"
 	lefthand_file = 'mojave/icons/mob/inhands/weapons/guns_inhand_left.dmi'
 	righthand_file = 'mojave/icons/mob/inhands/weapons/guns_inhand_right.dmi'
 	desc = "A modified A3-20 plasma caster built by REPCONN equipped with a multicasting kit that creates multiple weaker clots."
 	ammo_type = list(/obj/item/ammo_casing/energy/ms13/plasma/scatter)
-	cell_type = /obj/item/stock_parts/cell/ammo/mfc
+	cell_type = /obj/item/stock_parts/cell/ammo/pc
 
 /obj/item/gun/energy/ms13/laser/plasma/pistol
 	name ="MPL-A Plasma Pistol"
 	icon = 'mojave/icons/objects/guns/guns_world.dmi'
 	inhand_icon_state = "plasmapistol"
 	icon_state = "plasmapistol"
+	base_icon_state = "plasmapistol"
 	lefthand_file = 'mojave/icons/mob/inhands/weapons/guns_inhand_left.dmi'
 	righthand_file = 'mojave/icons/mob/inhands/weapons/guns_inhand_right.dmi'
 	desc = "A pistol-sized miniaturized plasma caster built by REPCONN. It fires heavy low penetration plasma clots."
 	ammo_type = list(/obj/item/ammo_casing/energy/ms13/plasma/pistol)
 	w_class = WEIGHT_CLASS_NORMAL
-	cell_type = /obj/item/stock_parts/cell/ammo/mfc
+	cell_type = /obj/item/stock_parts/cell/ammo/pc
 
 //projectiles
 
 /obj/projectile/beam/ms13
 	speed = 0.2 //Vanilla tg is 0.8
-
+	var/damage_constant = 0.8
+/obj/projectile/beam/ms13/Range()
+	if(hitscan != TRUE)
+		return ..()
+	var/turf/location = get_turf(src)
+	if(!location)
+		return ..()
+	var/datum/gas_mixture/environment = location.return_air()
+	var/environment_pressure = environment.return_pressure()
+	if(environment_pressure >= 50)
+		if((decayedRange - range) >= 4)
+			damage *= damage_constant
+	. = ..()
+/obj/projectile/beam/ms13/laser
+	name = "laser beam"
+	damage = 19
```

Comment:
You should be using the `extra_damage` and `extra_penetration` variables at the gun level instead of the projectile level for doing damage. This applies to all of the projectiles you're coding in. 

Also 19 damage is a huge meme but I assume this is just a w/e placeholder number.